internal/gatewayapi/translator.go

```diff
@@ -113,12 +126,24 @@ func (t *Translator) Translate(resources *Resources) *TranslateResult {
 	gateways := t.GetRelevantGateways(resources.Gateways)
 
 	// Process all Listeners for all relevant Gateways.
-	t.ProcessListeners(gateways, xdsIR, infraIR, resources)
+	t.ProcessListeners(gateways, nil, infraIR, resources)
+
+	return newTranslateInfraResult(gateways, infraIR)
+}
+
+func (t *Translator) TranslateXds(resources *Resources) *TranslateXdsResult {
+	xdsIR := &ir.Xds{}
+
+	// Get Gateways belonging to our GatewayClass.
+	gateways := t.GetRelevantGateways(resources.Gateways)
+
+	// Process all Listeners for all relevant Gateways.
+	t.ProcessListeners(gateways, xdsIR, nil, resources)
```

Comment:
does this mean we are computing `ProcessListeners` twice so we are computing the Gateway status twice ?

Reply:
Yes, see https://github.com/envoyproxy/gateway/issues/383.